src/modding/FullModDetails.cs

```diff
@@ -1,20 +1,61 @@
 ﻿using System;
+using Newtonsoft.Json;
 
 /// <summary>
 ///   An extended version of <see cref="ModInfo"/> that contains non-mod author controlled data
 /// </summary>
 public class FullModDetails : IEquatable<FullModDetails>
 {
+    [JsonConstructor]
```

Comment:
Needed to do that to stop mod creation from not working.

`Can't create mod due to parse failure on data: Newtonsoft.Json.JsonSerializationException: Unable to find a constructor to use for type FullModDetails. A class should either have a default constructor, one constructor with arguments or a constructor marked with the JsonConstructor attribute.`

Reply:
My point was that FullModDetails *should not be loaded from JSON*, as such it should not have a json constructor because it should *only* be constructed from code in a controlled manner. As I said this data should not be settable by the mod but set by the game code to be correct.